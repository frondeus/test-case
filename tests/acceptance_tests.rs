#![cfg(test)]

mod acceptance {
    use itertools::Itertools;
    use std::env;
    use std::path::PathBuf;
    use std::process::Command;

    #[test]
    fn basic() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("basic"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn hamcrest_assertions() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("hamcrest_assertions"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }

    #[test]
    fn r#async() {
        let output = Command::new("cargo")
            .current_dir(PathBuf::from("acceptance_tests").join("async"))
            .args(&["test"])
            .output()
            .expect("cargo command failed to start");

        let lines = String::from_utf8_lossy(&output.stdout)
            .to_string()
            .lines()
            .sorted()
            .join("\n");
        insta::assert_display_snapshot!(lines);
    }
}
