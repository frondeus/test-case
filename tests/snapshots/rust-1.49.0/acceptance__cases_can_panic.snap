---
source: tests/acceptance_tests.rs
assertion_line: 91
expression: output
---
      panic message: `"It has to panic"`,
    panicing::_expects_panicking_some_this_should_fail_
 expected substring: `"This should fail"`
---- panicing::_expects_panicking_some_this_should_fail_ stdout ----
error: test failed, to rerun pass '--lib'
failures:
failures:
running 5 tests
test panicing::_expects_panicking_some_it_has_to_panic_ ... ok
test panicing::_expects_panicking_some_this_should_fail_ ... FAILED
test panics_without_value::_expects_panicking_none ... ok
test pattern_matching_result_fails::simpleenum_var1_expects_matching_simpleenum_var2_ ... ok
test result: FAILED. 4 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
test result_which_panics::_2_2_expects_2_3 ... ok
thread 'panicing::_expects_panicking_some_this_should_fail_' panicked at 'It has to panic', src/lib.rs:20:5
